### PR TITLE
scala-cli: add page

### PR DESCRIPTION
**Pull request by xRuiAlves**
- [x] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [x] The page(s) have at most 8 examples.
- [x] The page description(s) have links to documentation or a homepage.
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- **Version of the command being documented (if known):** `1.0.4`

### CODE REVIEW

File: pages/common/scala-cli.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# scala-cli
+
+> Command-line tool to interact with the Scala language.
+> More information: <https://scala-cli.virtuslab.org/>.
```
Comment (pixelcmtd) [inline pages/common/scala-cli.md:4]:
```suggestion
> More information: <https://scala-cli.virtuslab.org/docs/overview/>.
```

File: pages/common/scala-cli.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+> Command-line tool to interact with the Scala language.
+> More information: <https://scala-cli.virtuslab.org/docs/overview/>.
+
+- Start a REPL (interactive shell) using a specific Scala and JVM version:
+
+`scala-cli --scala {{3.1.0}} --jvm {{temurin:17}}`
+
+- Compile and run a Scala script:
+
+`scala-cli run {{script.scala}}`
```
Comment (pixelcmtd) [inline pages/common/scala-cli.md:12]:
```suggestion
`scala-cli run {{path/to/script.scala}}`
```

File: pages/common/scala-cli.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+- Compile and test a Scala script:
+
+`scala-cli test {{script.scala}}`
```
Comment (pixelcmtd) [inline pages/common/scala-cli.md:16]:
```suggestion
`scala-cli test {{path/to/script.scala}}`
```

File: pages/common/scala-cli.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+- Format a Scala script:
+
+`scala-cli fmt {{script.scala}}`
```
Comment (pixelcmtd) [inline pages/common/scala-cli.md:20]:
```suggestion
`scala-cli fmt {{path/to/script.scala}}`
```

File: pages/common/scala-cli.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+- Generate files for IDE (VSCode and IntelliJ) support:
+
+`scala-cli setup-ide {{script.scala}}`
```
Comment (pixelcmtd) [inline pages/common/scala-cli.md:24]:
```suggestion
`scala-cli setup-ide {{path/to/script.scala}}`
```

**Review by pixelcmtd — COMMENTED** [review]
oops I missed that

File: pages/common/scala-cli.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
+# scala-cli
+
+> Command-line tool to interact with the Scala language.
```
Comment (kbdharun) [inline pages/common/scala-cli.md:3]:
```suggestion
> Tool to interact with the Scala language.
```
Ref. https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#descriptions

**Review by kbdharun — APPROVED** [review]
LGTM, Thanks for your contribution.

File: pages/common/scala-cli.md (hunk 6 above)
Comment (xRuiAlves) [reply pages/common/scala-cli.md:3]: Thanks! Makes sense

**Review by kbdharun — APPROVED** [review]

File: pages/common/scala-cli.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+- Compile and run a Scala script:
+
+`scala-cli run {{path/to/script.scala}}`
+
+- Compile and test a Scala script:
+
+`scala-cli test {{path/to/script.scala}}`
+
+- Format a Scala script:
```
Comment (gutjuri) [inline pages/common/scala-cli.md:18]: Does it format the file in-place or does it output the formatted contents to stdout?

**Review by gutjuri — CHANGES_REQUESTED** [review]

**Review by gutjuri — APPROVED** [review]
Thanks!

**Comment by xRuiAlves** [comment]
@pixelcmtd Did I cover all of your suggestions? 🤔

**Pull request merged**